plugins/techdocs/src/reader/components/TechDocsReaderPage/TechDocsReaderPage.tsx

```diff
@@ -143,6 +147,27 @@ export const TechDocsReaderLayout = (props: TechDocsReaderLayoutProps) => {
   );
 };
 
+function TechDocsAuthProvider({ children }: { children: ReactNode }) {
+  const app = useApp();
+  const { Progress } = app.getComponents();
+
+  const techdocsApi = useApi(techdocsApiRef);
+
+  const { loading, error } = useAsync(async () => {
```

Comment:
Might be best to have this be an effect instead with cleanup on unmount. Something like

```ts
useEffect(() => {
  const stopCookieRefresh = techdocsApi.startCookieRefresh()
  return () => {
    stopCookieRefresh()
  }
}, [techdocsApi])
```

Or the shorter version of that

Reply:
And in a future version you could also notify the user when refreshes fail consistently and expiry is hit.

Reply:
Ended up suggesting that we move the entire refresh mechanism to be managed as a React hook instead, so that it's easier to keep track of error states etc.

Reply:
The code is updated, could you both review again 🙏🏻 ?